### PR TITLE
chore(ci): add the Metrics feature to nox tests

### LINKED ISSUE #4521: Maintenance: add the Metrics feature to nox tests

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #4521 

### Changes

Adding the Metrics feature to nox tests

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Review by heitorlessa — DISMISSED** [review]
Q: should we only trust Nux test names to know what's required vs optional?

or **(a)** use a different test subfolder for each feature, **(b)** add a top-level docstring for those rearranged tests for optional dependency (like you did on noxfile:68

**Comment by leandrodamascena** [comment]
> Q: should we only trust Nux test names to know what's required vs optional?

I think that's a good indication of what dependencies this subset of tests should use, no?
> 
> or **(a)** use a different test subfolder for each feature, **(b)** add a top-level docstring for those rearranged tests for optional dependency (like you did on noxfile:68

Can you elaborate further and give an example of what this would look like?

**Comment by heitorlessa** [comment]
Of course!

We can reduce cognitive load on maintainers by making this even more explicit with a folder name that suggests the reason behind this separation.

For example.

* **Now**: `tests/functional/metrics`
* **After**:
  * `tests/functional/metrics/required_dependencies/` (CloudWatch EMF, no dependency really)
  * `tests/functional/metrics/datadog`
  * `tests/functional/metrics/otel`

Alternatively, you could bundle `/datadog, /otel` under `dependencies`, but the trade-off is that you lose the ability to selectively test that `datadog` dependency only.

As a byproduct, this also helps us detect regressions that were introduced in one dependency that now affects another unrelated class of tests.

**Comment by leandrodamascena** [comment]
> For example.
> 
> * **Now**: `tests/functional/metrics`
> * **After**:
>   
>   * `tests/functional/metrics/required_dependencies/` (CloudWatch EMF, no dependency really)
>   * `tests/functional/metrics/datadog`
>   * `tests/functional/metrics/otel`
> 

This is much better @heitorlessa! Thanks for always making things simple ❤️

File: noxfile.py [hunk 1]
```diff
@@ -50,9 +50,25 @@ def build_and_run_test(session: nox.Session, folders: List, extras: Optional[str
...
     """Tests that only depends for required libraries"""
     # Logger
+    # Metrics - Amazon CloudWatch EMF
+    # Metrics - Base provider
     build_and_run_test(
         session,
         folders=[
-            f"{PREFIX_TESTS_FUNCTIONAL}/logger/",
+            f"{PREFIX_TESTS_FUNCTIONAL}/logger/required_dependencies/",
+            f"{PREFIX_TESTS_FUNCTIONAL}/metrics/required_dependencies/",
```
Comment (heitorlessa) [inline noxfile.py:59]: see how beautiful it is now - both in Nox and on folder structure <3 no ambiguity now!

**Review by heitorlessa — APPROVED** [review]

**Pull request merged**